### PR TITLE
Merge apache/v1.6.3 into branch criteo-1.6 and validate with Travis CI

### PR DESCRIPTION
**Pull request by AnthonyTruchet**
## What changes were proposed in this pull request?

Merge apache/v1.6.3 into branch criteo-1.6. (Duplicate with #12)

## How was this patch tested?

Validate with Travis CI: https://travis-ci.org/criteo-forks/spark

### CODE REVIEW

File: dev/criteo/test-with-maven.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# You can run tests for a module only by specifying -pl <module>
+# as argument to this script, or some specific test by passing
+# -Dtest=... (see surefire documentation).
+
+source `dirname "$0"`/criteo-spark-env.sh
+
+export _JAVA_OPTIONS="-Xss2048k -Dspark.buffer.pageSize=1048576 -Xmx4g"
```
Comment (ashangit) [inline dev/criteo/test-with-maven.sh:9]: Do we really need those parameters 4g seems quite a lot and Xss at 2048k will take also more mem per threads.
By the way those settings should be set for maven: export MAVEN_OPTS="-Xmx2g -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=512m"

Comment (AnthonyTruchet) [reply dev/criteo/test-with-maven.sh:9]: I do not know, I just replayt he setting one spark developper published on the sparl-dev mailing list. Feel free to adjust - or I'll do it next push

Comment (ashangit) [reply dev/criteo/test-with-maven.sh:9]: Please remove any commits related to TravisCI on this MR.
So far it seems that ressources available on Travis to do the build/test are not sufficient.
We should study this before.

Comment (AnthonyTruchet) [reply dev/criteo/test-with-maven.sh:9]: Agreed

**Comment by AnthonyTruchet** [comment]
PR abandonned and replaced by #15

**Pull request closed without merging**